lib/tapjoy/ldap/base.rb

```diff
@@ -8,12 +8,16 @@ class Base
       def initialize
         ldap_config_file = "#{ldap_config_directory}/ldap_info.yaml"
         ldap_password_file = "#{ldap_config_directory}/ldap.secret"
-        @ldap_info    = YAML.load_file(ldap_config_file)
-        @hosts        = @ldap_info['servers']
-        @basedn       = @ldap_info['basedn']
-        @conn         = find_valid_host(ldap_password_file)
-        @service_ou   = @ldap_info['service_ou']
-        @email_domain = @ldap_info['email_domain']
+
+        if File.readable?(ldap_config_file) && File.readable?(ldap_password_file)
+          load_config_from_files(ldap_config_file, ldap_password_file)
+        else
+          begin
```

Comment:
it is unclear to me why we have a begin / rescue block around `load_config_from_env` but not `load_config_from_files`?

Reply:
Because if we can't load from files, we can failover to the ENV.  If we can't load from the ENV, there's no fallback.

Reply:
where does the exception get thrown?

Reply:
I just revamped the exception handling. I hope it's a bit cleaner now